go/test/endtoend/tabletgateway/healthcheck/vtgate_test.go

```diff
@@ -169,9 +178,14 @@ func isMasterTabletPresent(tablets map[string]interface{}) bool {
 	return false
 }
 
-func exec(t *testing.T, conn *mysql.Conn, query string) *sqltypes.Result {
+func exec(t *testing.T, conn *mysql.Conn, query string, expectError string) *sqltypes.Result {
 	t.Helper()
 	qr, err := conn.ExecuteFetch(query, 1000, true)
-	require.Nil(t, err)
+	if expectError == "" {
+		require.Nil(t, err)
```

Comment:
When checking for errors, we get better assertion error messages is we use `require.NoError` instead of `require.Nil`

Reply:
+1. I was looking for this, but thought it was something like NilError.